specs/MultiProfile.md

```diff
@@ -363,16 +384,36 @@ interface ICoreWebView2Profile2 : ICoreWebView2Profile {
   [propget] HRESULT CookieManager([out, retval] ICoreWebView2CookieManager** cookieManager);
 }
 
-[uuid(1c1ae2cc-d5c2-ffe3-d3e7-7857035d23b7), object, pointer_default(unique)]
-interface ICoreWebView2Profile3 : ICoreWebView2Profile2 {
-  /// All webviews on this profile will be closed, and the profile will be marked for deletion.
-  /// After the Delete() call completes, The render process of webviews on this profile will
-  /// asynchronously exit with the reason:`COREWEBVIEW2_PROCESS_FAILED_REASON_PROFILE_DELETED`.
-  /// See 'COREWEBVIEW2_PROCESS_FAILED_REASON::COREWEBVIEW2_PROCESS_FAILED_REASON_PROFILE_DELETED'
-  /// for more details. The profile directory on disk will be actually deleted when the browser
-  /// process exits. Webview2 creation will fail with the HRESULT is ERROR_INVALID_STATE(0x8007139FL)
-  /// if you create it with the same name as a profile that is being deleted.
-  HRESULT Delete(); 
+[uuid(2765B8BD-7C57-4B76-B8AA-1EC940FE92CC), object, pointer_default(unique)]
+interface ICoreWebView2StagingProfile4 : IUnknown {
+  /// After the API is called, the profile will be marked for deletion. The
+  /// local profile’s directory will be tried to delete at browser process
+  /// exit, if fail to delete, it will recursively try to delete at next
+  /// browser process start until successful.
+  /// The corresponding user's `ProfileDeleted` event handle function will
+  /// be triggered. After the function is triggered, continuing to use the
+  /// profile or its corresponding webviews is an undefined behavior.
+  /// For each WebView of the same profile, the WebView will be auto closed.
+  /// If the user has registered an event handler for ProfileDeleted event,
+  /// the event handle will be invoked before WebView is closed.
+  /// If create a new profile with the same name as the profile that has been
+  /// marked as deleted will be failure with the HRESULT:ERROR_INVALID_STATE
+  /// (0x8007139FL).
+  HRESULT Delete();
+}
+
+[uuid(cc39bea3-f6f8-471b-919f-fa253e2fff03), object, pointer_default(unique)]
+interface ICoreWebView2Staging9 : IUnknown {
+  /// The `ProfileDeleted` event is raised when its corresponding Profile's
+  /// Delete API is called. When this event has been raised, continue to use
```

Comment:
The text for Delete above says that this event is raised before the webview2 is closed which means I'm guessing that its also before the profile has been deleted. But this text does not make it clear if the event is raised before or after deletion. Is it OK to use the WebView2 and the Profile object in the event handler or not?

If the profile is deleted after raising the event, and its OK to use the WebView2 and Profile objects, then this event should be named ProfileDeletionStarting. And we should say in this documentation that it is OK to use the CoreWebView2Profile object and corresponding WebView2 objects in the event handler but they will be closed after and it is not safe to use them after.

If the event is raised after the profile is deleted and the objects are closed, then the documentation in CoreWebView2Profile.Delete needs to be updated.

Reply:
The event is raised before deletion. In the user's event handle function, the webview2 and profile object has not been released, they are still alive, but the deletion is async, some flag has been set, although the objects are still alive, we should not allow users to continue using them.

In the user's event handle function, continue using the webview2's API, most can work normally, but some of them can't work properly. And Delete API is a tough behaviour, if the user has called this API, they should not continue using this profile and its corresponding webview2. If all webview2s in a same client process, this event handle is not needed, but some webview2s may be in different process, so, I add an event to notify the user that the webview2 is invalid and should not use it anymore.

So, I document it: continue to use the profile or its corresponding webviews is an undefined behaviour.

Reply:
So the event is raised when the profile is marked to be deleted. Can we leave everything else the same but close the profile and associated webview objects before raising the event so things are in a consistent state when the event is raised? End developers shouldn't be trying to use these objects during the event because the profile is either deleted or soon to be deleted and its better if we can express that to the end developer at runtime by having the objects closed rather than requiring the end developer to read the documentation and know from that not to try using the objects.

Reply:
Ok, have change the logic to:
1. Profile.Delete API called
2. Profile marked as deleted.
3. auto close the webview2.
4. raise CoreWebview2.Close event